prodactivity-frontend/managers/AuthenticationManager.tsx

```diff
@@ -4,29 +4,56 @@ import KeychainHelper from '../helpers/KeychainHelper';
 import { TokenType } from '../application/Enums';
 
 export default class AuthenticationManager {
-
     baseURL = 'http://localhost:5000/authentication';
 
-    constructor() {
-    }
+    constructor() {}
 
     /**
      * @param {string} email The user's email address for sign in
      * @param {string} password The user's password for sign in
-     * 
+     *
      * @returns {AuthenticationViewModel} View Model containing all info for the login
      */
     async signIn(email: string, password: string): Promise<Result<AuthenticationViewModel>> {
         try {
-            let response = await FetchHelper.post(`${this.baseURL}/signIn`, {"email": email, "password": password});
+            let response = await FetchHelper.post(`${this.baseURL}/signIn`, { email: email, password: password });
+            if (response) {
+                var authViewModel: AuthenticationViewModel = Object.assign(new AuthenticationViewModel(), response);
+                await this.storeAuthenticationTokens(authViewModel);
+                return { result: authViewModel };
+            }
+
+            return { message: 'There has been a problem signing in, try a different email/password and try again.' };
+        } catch (error) {
+            console.log(error);
+            return { message: error.message };
+        }
+    }
 
+    /**
+     * @param {string} email The user's email address for sign up
+     * @param {string} password The user's password for sign up
+     * @param {string} firstName The user's first name
+     * @param {string} lastName The user's last name
+     *
+     * @returns {AuthenticationViewModel} View Model containing all info for the sign up
+     */
+    async signUp(
+        email: string,
+        password: string,
+        firstName: string,
+        lastName: string,
+    ): Promise<Result<AuthenticationViewModel>> {
+        try {
+            let responseBody = { email: email, password: password, firstName: firstName, lastName: lastName };
```

Comment:
Minor thing, but I think we should use 'const' whenever a variable doesn't get re-assigned, and 'let' if it does instead of var.